### PR TITLE
Allow assigning meeting license on synced users

### PR DESCRIPTION
**Pull request by baumann-tv**
Add new configuration property that specifies a Guid that represents a
valid TeamViewer Meeting license activation code. If set, the license
will automatically be activated on newly created company members.

Note: For this to work the license must already be activated on the
company profile and it must still have free organizer license slots,
else the set license will be silently ignored.

* Add optional 'MeetingLicenseKey' property to configuration.
* Forward set meeting license key to create member Web API call

### CODE REVIEW

File: TeamViewerADConnector/Internal/Configuration.ps1 [hunk 1]
```diff
@@ -48,4 +49,8 @@ function Confirm-Configuration($config) {
     if ($config.UseSsoCustomerId -And [string]::IsNullOrWhiteSpace($config.SsoCustomerId)) {
         Throw "The parameter 'SsoCustomerId' cannot be empty if 'UseSsoCustomerId' is configured."
     }
+	$parseGuid = [System.Guid]::empty
```
Comment (danieljoos) [inline TeamViewerADConnector/Internal/Configuration.ps1:52]: The check can be simplified, e.g. like this:
```suggestion
![string]::IsNullOrEmpty($meetingKey) -And [guid]$meetingKey | Out-Null
```

Also: tabs vs. spaces. I suggest using vscode + Powershell Extension to format the files

File: TeamViewerADConnector/Internal/Sync.ps1 [hunk 2]
```diff
@@ -195,9 +195,13 @@ function Invoke-SyncUser($syncContext, $configuration, $progressHandler) {
                 if ($configuration.UseSsoCustomerId) {
                     $newUser.sso_customer_id = $configuration.SsoCustomerId;
                 }
+                if ($configuration.MeetingLicenseKey) {
+                    $newUser.meeting_license_key = $configuration.MeetingLicenseKey;
+                }
                 try {
                     $addedUser = (Add-TeamViewerUser $configuration.ApiToken $newUser)
                     $newUser.id = $addedUser.id
+                    $newUser.MeetingLicenseKey = $addedUser.activated_meeting_license_key
```
Comment (danieljoos) [inline TeamViewerADConnector/Internal/Sync.ps1:204]: why is this necessary? The data is not shown anywhere in the logs, isn't it?
If really needed, I'd suggest to rename this to `ActivatedMeetingLicenseKey`

File: Tests/TeamViewerADConnector/Internal/Sync.Tests.ps1 [hunk 3]
```diff
@@ -248,6 +248,60 @@ Describe 'Invoke-SyncUser' {
...
+
+        It 'Should set meeting license on new user passed to Add-TeamViewerUser' {
+            Mock Add-TeamViewerUser { }
+
+            Invoke-SyncUser $syncContext $configuration { }
+            Assert-MockCalled Add-TeamViewerUser -Times 1 -Scope It `
+                -ParameterFilter { $user -And $user.meeting_license_key -eq $configuration.MeetingLicenseKey }
+        }
+
+        It 'Should not be set non-existant meeting license on new user passed to Add-TeamViewerUser' {
```
Comment (danieljoos) [inline Tests/TeamViewerADConnector/Internal/Sync.Tests.ps1:277]: small typo in the description:
```
non-existant
         ^
```

**Review by danieljoos — COMMENTED** [review]
Just 1-2 small things. otherwise looks very good to me

File: Tests/TeamViewerADConnector/Internal/Sync.Tests.ps1 (hunk 3 above)
Comment (baumann-tv) [reply Tests/TeamViewerADConnector/Internal/Sync.Tests.ps1]: Fixed

File: TeamViewerADConnector/Internal/Configuration.ps1 (hunk 1 above)
Comment (baumann-tv) [reply TeamViewerADConnector/Internal/Configuration.ps1]: That definitely feels less clunky. (Though I did add a comment just to clear why/what is happening.)

File: TeamViewerADConnector/Internal/Sync.ps1 (hunk 2 above)
Comment (baumann-tv) [reply TeamViewerADConnector/Internal/Sync.ps1]: I somehow ended up with the impression, that the user data is returned to the caller.
As it's actually not used at the moment I have removed this line (and adapted the test) .. but I will think about potentially adding a log entry for cases where the assignment failed.

File: TeamViewerADConnector/Internal/Configuration.ps1 (hunk 1 above)
Comment (baumann-tv) [reply TeamViewerADConnector/Internal/Configuration.ps1]: Regarding the tabs: That one slipped through. There should not be any left now.
